src/functions/marshal_headers_from_sapi.php

```diff
@@ -35,18 +44,14 @@ function marshalHeadersFromSapi(array $server) : array
             }
         }
 
-        if ($value === '') {
-            continue;
-        }
-
         if (strpos($key, 'HTTP_') === 0) {
-            $name = strtr(strtolower(substr($key, 5)), '_', '-');
+            $name = str_replace('_', '-', strtolower(substr($key, 5)));
             $headers[$name] = $value;
             continue;
         }
 
         if (strpos($key, 'CONTENT_') === 0) {
-            $name = 'content-' . strtolower(substr($key, 8));
+            $name = str_replace('_', '-', strtolower($key));
```

Comment:
Why have you changed this? It's unrelated...

Reply:
See commits, I've written test first for the whole behaviour of the method I would expect, and then I've made a changes to fulfil test. Yes, it seems to be separate issue.

Reply:
When we have multiple `_` in the key - it is possible for `HTTP_*` headers, but not sure now if for `CONTENT_*`. I know Content-Security-Policy or Content-Transfer-Encoding, but these are response headers, not request. So maybe it is not a case?

Reply:
Hmm, that's interesting. My understanding was that `strtr()` would replace all instances - and a quick test of the following shows it definitely does:

```php
$string = 'CONTENT_SECURITY_POLICY';
$revised = strtr('CONTENT_SECURITY_POLICY', '_', '-'); // 'CONTENT-SECURITY-POLICY'
```

(see https://3v4l.org/uEKEQ for an example; I also do `strtolower()` in that one)

Reply:
Perhaps it's the `substr()` call we need to remove instead?

Reply:
@weierophinney there was no strtr in that line ...
I've removed concatenation and substr and added str_replace

Reply:
So, since it's new, let's go with `strtr()`, as that would make it consistent with the previous usage.

Reply:
Updated